src/lib.rs

```diff
@@ -250,72 +323,149 @@ impl UinputAbsSetup {
+    /// }
+    /// ```
     #[inline]
     pub fn kind(&self) -> InputEventKind {
-        let code = self.code();
-        match self.event_type() {
-            EventType::SYNCHRONIZATION => InputEventKind::Synchronization(Synchronization(code)),
-            EventType::KEY => InputEventKind::Key(Key::new(code)),
-            EventType::RELATIVE => InputEventKind::RelAxis(RelativeAxisType(code)),
-            EventType::ABSOLUTE => InputEventKind::AbsAxis(AbsoluteAxisType(code)),
-            EventType::MISC => InputEventKind::Misc(MiscType(code)),
-            EventType::SWITCH => InputEventKind::Switch(SwitchType(code)),
-            EventType::LED => InputEventKind::Led(LedType(code)),
-            EventType::SOUND => InputEventKind::Sound(SoundType(code)),
-            EventType::FORCEFEEDBACK => InputEventKind::ForceFeedback(code),
-            EventType::FORCEFEEDBACKSTATUS => InputEventKind::ForceFeedbackStatus(code),
-            EventType::UINPUT => InputEventKind::UInput(code),
-            _ => InputEventKind::Other,
+        match self {
+            InputEvent::Synchronization(ev) => InputEventKind::Synchronization(ev.kind()),
+            InputEvent::Key(ev) => InputEventKind::Key(ev.kind()),
+            InputEvent::RelativeAxis(ev) => InputEventKind::RelativeAxis(ev.kind()),
+            InputEvent::AbsoluteAxis(ev) => InputEventKind::AbsoluteAxis(ev.kind()),
+            InputEvent::Misc(ev) => InputEventKind::Misc(ev.kind()),
+            InputEvent::Switch(ev) => InputEventKind::Switch(ev.kind()),
+            InputEvent::Led(ev) => InputEventKind::Led(ev.kind()),
+            InputEvent::Sound(ev) => InputEventKind::Sound(ev.kind()),
+            InputEvent::Repeat(ev) => InputEventKind::Repeat(ev.kind()),
+            InputEvent::ForceFeedback(ev) => InputEventKind::ForceFeedback(ev.kind()),
+            InputEvent::Power(ev) => InputEventKind::Power(ev.kind()),
+            InputEvent::ForceFeedbackStatus(ev) => InputEventKind::ForceFeedbackStatus(ev.kind()),
+            InputEvent::UInput(ev) => InputEventKind::UInput(ev.kind()),
+            InputEvent::Other(ev) => InputEventKind::Other(ev.kind()),
         }
     }
 
-    /// Returns the raw "value" field directly from input_event.
+    /// A convenience function to return the `InputEvent` its `kind()` and `value()` wrapped in a
+    /// certain newtype corresponding to the `InputEvent` variant.
     ///
-    /// For keys and switches the values 0 and 1 map to pressed and not pressed respectively.
-    /// For axes, the values depend on the hardware and driver implementation.
+    /// # Example
+    /// ```
+    /// use evdev::*;
+    /// let event =  InputEvent::new(1, KeyType::KEY_A.0, 1);
+    /// match event.matcher() {
+    ///     InputEventMatcher::Key(KeyEvent, KeyType::KEY_A, 1) => (),
+    ///     _=> panic!(),
+    /// }
+    /// ```
     #[inline]
-    pub fn value(&self) -> i32 {
-        self.0.value
+    pub fn matcher(self) -> InputEventMatcher {
```

Comment:
maybe a better name for this method is `decode`?

Reply:
maybe. I am actually not super happy with this currently. It feels a bit much that there are three places where the EventType is matched 1. the `InputEventMatcher` variant, 2. the `InputEvent` variant and 3. the `...Type` type. I will try a few different things with that and see if there is a better solution.

This also makes the `InputEventKind` enum redundant. 
I would like to remove the `InputEventKind` I didn't do that yet to not break the interface to much. But I really don't see any reason to keep it. What do you think about that?

Reply:
i'm generally happy to see things like that collapsed out if the API